### ISSUE TITLE
How to set options to mimalloc before mi_process_init

### ISSUE THREAD

**Issue by santagada** [microsoft/mimalloc#433]
Maybe is my misunderstanding of mimalloc but most options you would want to set before mi_process_init runs (like to reserve memory and etc) how to accomplish that? No matter where I try to put the options it seems like mi_process_init had already happened.

Maybe its related to the difference between mi_option_set and mi_option_set_default, which is also not explained in the docs.

**Comment by romange** [comment]
You can use environment variables https://microsoft.github.io/mimalloc/environment.html  if you use mimalloc as your main allocator. Also, most options should work even when you change them in the middle of the run since they are checked directly. 
for example, https://github.com/microsoft/mimalloc/blob/8a1f8a305adb5708935a445ba2a5daf9961a4466/src/os.c#L122

**Comment by santagada** [comment]
Yes and no, the specific one I mention reserve memory I think needs to be set before the init, also mi_option_large_os_pages would be good to be set before the init to guarantee everything is done with large_os_pages.
Setting env vars would work, except it is not internal to the application, or do you mean I go, parse the env vars that might be set and override the env vars before mi_process_init runs?
